source/common/http/conn_manager_utility.cc

```diff
@@ -439,6 +439,10 @@ bool ConnectionManagerUtility::maybeNormalizePath(RequestHeaderMap& request_head
     return true; // It's as valid as it is going to get.
   }
   bool is_valid_path = true;
+  config.normalizePath(request_headers);
+  if (!config.shouldNormalizePath() && !config.shouldMergeSlashes()) {
```

Comment:
You can implement the old normalization options through the new mechanism. In this case you will not need this code here. See more on it below.